src/bindings/c/include/openvino/ov_property.h

```diff
@@ -0,0 +1,150 @@
+                                       //!<  hint for a range for number of async infer requests. If device supports
+                                       //!<  streams, the metric provides range for number of IRs per stream.
+    RANGE_FOR_STREAMS,  //!<  Read-only property<unsigned int, unsigned int> to provide information about a range for
+                        //!<  streams on platforms where streams are supported
+    FULL_DEVICE_NAME,   //!<  Read-only property<char *> to get a string value representing a full device name.
+    OPTIMIZATION_CAPABILITIES,  //!<  Read-only property<char *> to get a string list of capabilities options per
+                                //!<  device.
+    CACHE_DIR,    //!<  Read-write property<char *> to set/get the directory which will be used to store any data cached
+                  //!<  by plugins.
+    NUM_STREAMS,  //!<  Read-write property<uint32_t> to set/get the number of executor logical partitions
+    AFFINITY,  //!<  Read-write property<ov_affinity_e> to set/get the name for setting CPU affinity per thread option.
+    INFERENCE_NUM_THREADS,  //!<  Read-write property<int32_t> to set/get the maximum number of threads that can be used
+                            //!<  for inference tasks.
+    PERFORMANCE_HINT,       //!< Read-write property<ov_performance_mode_e>, it is high-level OpenVINO Performance Hints
+                       //!< unlike low-level properties that are individual (per-device), the hints are something that
+                       //!< every device accepts and turns into device-specific settings detail see
+                       //!< ov_performance_mode_e to get its hint's key name
+    NETWORK_NAME,              //!<  Read-only property<char *> to get a name of name of a model
+    INFERENCE_PRECISION_HINT,  //!< Read-write property<ov_element_type_e> to set the hint for device to use specified
+                               //!< precision for inference
+    OPTIMAL_BATCH_SIZE,  //!<  Read-only property<uint32_t> to query information optimal batch size for the given device
+                         //!<  and the network
+    MAX_BATCH_SIZE,  //!<  Read-only property to get maximum batch size which does not cause performance degradation due
+                     //!<  to memory swap impact.
+    PERFORMANCE_HINT_NUM_REQUESTS,  //!<  (Optional) property<uint32_t> that backs the Performance Hints by giving
+                                    //!<  additional information on how many inference requests the application will be
+                                    //!<  keeping in flight usually this value comes from the actual use-case  (e.g.
+                                    //!<  number of video-cameras, or other sources of inputs)
+} ov_property_key_e;
+
+/**
+ * @enum ov_property_value_type_e
+ * @brief Enum to define property value type.
+ */
+typedef enum {
+    BOOL = 0U,  //!< boolean data
+    CHAR,       //!< char data
+    ENUM,       //!< enum data
```

Comment:
What will we do in case of enum in property?

Reply:
enum will define the data size to be 32bit, so that we can figure out the property data size.

Reply:
It is not fully true, because I can create enum with different size. Moreover user cannot use this information in order to cast value to some specific enum.
Do we really need this value at the current moment? Or we can remove it for now and introduce in the future if it is required.

Reply:
I think we do need enum due to some property value are enumerate, such as  performance_mode, affinity and element type.  And such enum are defined by us, we can control its size.